File: workshops/agentcore-workday-integration/dashboard-gradio/callback_router.py
import logging
from urllib.parse import unquote
from fastapi import APIRouter, Request
from fastapi.responses import RedirectResponse
from ac_client import invoke_agent

l = logging.getLogger("aws.callback")

router = APIRouter()


@router.get("/app/callback")
async def callback(request: Request):
    session_id = unquote(request.query_params.get("session_id", ""))
    l.info(f"> callback session_id={session_id}")

    payload = {"cmd": "completeAuth", "session_id": session_id}
    response = invoke_agent(payload)
    l.info(f"> init_agent response={response}")
    return RedirectResponse(url="/app")

File: workshops/agentcore-workday-integration/dashboard-js/callback_router.js
import { Router } from "express";
import { invokeAgent } from "./ac_client.js";

const router = Router();

router.get("/app/callback", async (req, res) => {
  const sessionId = decodeURIComponent(req.query.session_id || "");
  console.log(`> callback session_id=${sessionId}`);
  try {
    await invokeAgent({ cmd: "completeAuth", session_id: sessionId });
  } catch (e) {
    console.error("completeAuth error", e);
  }
  res.redirect("/app");
});

export default router;
